### PR TITLE
fix(dtfs2-7063): facility id record are missing in acbs

### PR DESCRIPTION
**Pull request by AlexBramhill**
## Introduction :pencil2:
Azure functions were failing to create deals. This was due to an issue with dates, and how we parse them.

When we create a deal in azure functions we pass `submissionDate`, which is an `epoch`, through `formatDate` returning an `iso date time as string`. Later in the deal creation process in azure functions, we call the same `formatDate` function on this newly created `iso date time as string`. This caused an `invalidDate` to be created, and lead to the deal failing to create.

We would expect this `iso date time as string` to return a `formatted date` when parsed through `formatDate`

## Resolution :heavy_check_mark:
We added handling to allow `formatDate` to parse `iso date time as sting`s

## Miscellaneous :heavy_plus_sign:
We've added test cases to ensure this is fixed, as well as running the data from the bug report locally to ensure it returns expected values.

### CODE REVIEW

File: azure-functions/acbs-function/helpers/date.js [hunk 1]
```diff
@@ -150,13 +152,7 @@ const getDateStringFromYearMonthDay = (year, month, day) => {
...
   }
 
-  return formatOrReturnInvalidDate(
-    new Date(
-      Number(formatYear(year)),
-      Number(month) - 1,
-      Number(day),
-    ),
-  );
+  return formatOrReturnInvalidDate(new Date(Number(formatYear(year)), Number(month) - 1, Number(day)));
```
Comment (abhi-markan) [inline azure-functions/acbs-function/helpers/date.js:155]: Can the argument computation be moved to `variable` or a `function` to improve readbility?

**Review by abhi-markan — CHANGES_REQUESTED** [review]

File: azure-functions/acbs-function/helpers/date.js (hunk 1 above)
Comment (AlexBramhill) [reply azure-functions/acbs-function/helpers/date.js]: added -- sorry for rebase

File: azure-functions/acbs-function/helpers/date.js [hunk 2]
```diff
@@ -144,19 +146,20 @@ const getDateStringFromYearMonthDay = (year, month, day) => {
...
 
-  const daysInMonth = getDaysInMonth(new Date(Number(formatYear(year)), Number(month) - 1));
+  const yearAsNumber = Number(formatYear(year));
+  const monthAsZeroIndexedNumber = Number(month) - 1;
+  const dayAsNumber = Number(day);
+  const startOfMonth = new Date(yearAsNumber, monthAsZeroIndexedNumber);
+
+  const daysInMonth = getDaysInMonth(startOfMonth);
 
   if (daysInMonth < Number(day)) {
```
Comment (ChristianMcc) [inline azure-functions/acbs-function/helpers/date.js:156]: This could technically be dayAsNumber

**Review by ChristianMcc — COMMENTED** [review]
lil nit pick

**Review by ChristianMcc — APPROVED** [review]
a green tile for me

**Review by abhi-markan — APPROVED** [review]

**Pull request merged**